### ISSUE TITLE
[Bug] BE crash on SegmentIterator

### ISSUE THREAD

**Issue by morningman** [apache/incubator-doris#5609]
**Describe the bug**
BE crash with error:

![image](https://user-images.githubusercontent.com/2899462/113886831-8fc79180-97f3-11eb-894e-17a464c71f05.png)

**To Reproduce**

Hard to reproduce.
The direct reason is because the read parameters passed by the BE side does not contain columns
related to the deletion conditions, which results in the corresponding column iterator being null when
preparing to filter the data according to the deletion conditions.

The root cause is that the FE side does not correctly set the `isPreAggregate` attribute during query planning,
which causes the BE side to skip the delete condition column by mistake.

 - Version [0.14.7]

**Comment by qidaye** [comment]
To facilitate the search for people who encounter this bug.

```c++
*** Check failure stack trace: ***
    @          0x1bd92dd  google::LogMessage::Fail()
    @          0x1bdb164  google::LogMessage::SendToLog()
    @          0x1bd8e04  google::LogMessage::Flush()
    @          0x1bdbb99  google::LogMessageFatal::~LogMessageFatal()
    @           0xf8cd34  doris::CondColumn::del_eval()
    @          0x17145d4  doris::segment_v2::ColumnReader::_get_filtered_pages()
    @          0x1714c48  doris::segment_v2::FileColumnIterator::get_row_ranges_by_zone_map()
    @          0x16eaaaa  doris::segment_v2::SegmentIterator::_get_row_ranges_from_conditions()
    @          0x16ebdf8  doris::segment_v2::SegmentIterator::_get_row_ranges_by_column_conditions()
    @          0x16ec350  doris::segment_v2::SegmentIterator::_init()
    @          0x16ecbce  doris::segment_v2::SegmentIterator::next_batch()
    @           0xfde23c  doris::BetaRowsetReader::next_block()
    @           0xf5b9b7  doris::CollectIterator::Level0Iterator::_refresh_current_row()
    @           0xf5bd12  doris::CollectIterator::add_child()
    @           0xf3a11e  doris::Reader::_capture_rs_readers()
    @           0xf3e6db  doris::Reader::init()
    @           0xf29800  doris::Merger::merge_rowsets()
    @           0xf18720  doris::Compaction::do_compaction_impl()
    @           0xf1ae4c  doris::Compaction::do_compaction()
    @           0xf0f7ec  doris::BaseCompaction::execute_compact_impl()
    @           0xf17004  doris::Compaction::execute_compact()
    @           0xe9f9ab  doris::Tablet::execute_compaction()
    @           0xe79c3f  _ZNSt17_Function_handlerIFvvEZN5doris13StorageEngine35_compaction_tasks_producer_callbackEvEUlvE0_E9_M_invokeERKSt9_Any_data
    @          0x120ac92  doris::ThreadPool::dispatch_thread()
    @          0x1202af8  doris::Thread::supervise_thread()
    @     0x7fddd1716ea5  start_thread
    @     0x7fddd1a298dd  __clone
```

**Comment by acelyc111** [comment]
> To facilitate the search for people who encounter this bug.
> 
> ```c++
> *** Check failure stack trace: ***
>     @          0x1bd92dd  google::LogMessage::Fail()
>     @          0x1bdb164  google::LogMessage::SendToLog()
>     @          0x1bd8e04  google::LogMessage::Flush()
>     @          0x1bdbb99  google::LogMessageFatal::~LogMessageFatal()
>     @           0xf8cd34  doris::CondColumn::del_eval()
>     @          0x17145d4  doris::segment_v2::ColumnReader::_get_filtered_pages()
>     @          0x1714c48  doris::segment_v2::FileColumnIterator::get_row_ranges_by_zone_map()
>     @          0x16eaaaa  doris::segment_v2::SegmentIterator::_get_row_ranges_from_conditions()
>     @          0x16ebdf8  doris::segment_v2::SegmentIterator::_get_row_ranges_by_column_conditions()
>     @          0x16ec350  doris::segment_v2::SegmentIterator::_init()
>     @          0x16ecbce  doris::segment_v2::SegmentIterator::next_batch()
>     @           0xfde23c  doris::BetaRowsetReader::next_block()
>     @           0xf5b9b7  doris::CollectIterator::Level0Iterator::_refresh_current_row()
>     @           0xf5bd12  doris::CollectIterator::add_child()
>     @           0xf3a11e  doris::Reader::_capture_rs_readers()
>     @           0xf3e6db  doris::Reader::init()
>     @           0xf29800  doris::Merger::merge_rowsets()
>     @           0xf18720  doris::Compaction::do_compaction_impl()
>     @           0xf1ae4c  doris::Compaction::do_compaction()
>     @           0xf0f7ec  doris::BaseCompaction::execute_compact_impl()
>     @           0xf17004  doris::Compaction::execute_compact()
>     @           0xe9f9ab  doris::Tablet::execute_compaction()
>     @           0xe79c3f  _ZNSt17_Function_handlerIFvvEZN5doris13StorageEngine35_compaction_tasks_producer_callbackEvEUlvE0_E9_M_invokeERKSt9_Any_data
>     @          0x120ac92  doris::ThreadPool::dispatch_thread()
>     @          0x1202af8  doris::Thread::supervise_thread()
>     @     0x7fddd1716ea5  start_thread
>     @     0x7fddd1a298dd  __clone
> ```

Seems this bug is not same with morningman's? What version/commit id you are using?

**Comment by qidaye** [comment]
> > To facilitate the search for people who encounter this bug.
> > ```c++
> > *** Check failure stack trace: ***
> >     @          0x1bd92dd  google::LogMessage::Fail()
> >     @          0x1bdb164  google::LogMessage::SendToLog()
> >     @          0x1bd8e04  google::LogMessage::Flush()
> >     @          0x1bdbb99  google::LogMessageFatal::~LogMessageFatal()
> >     @           0xf8cd34  doris::CondColumn::del_eval()
> >     @          0x17145d4  doris::segment_v2::ColumnReader::_get_filtered_pages()
> >     @          0x1714c48  doris::segment_v2::FileColumnIterator::get_row_ranges_by_zone_map()
> >     @          0x16eaaaa  doris::segment_v2::SegmentIterator::_get_row_ranges_from_conditions()
> >     @          0x16ebdf8  doris::segment_v2::SegmentIterator::_get_row_ranges_by_column_conditions()
> >     @          0x16ec350  doris::segment_v2::SegmentIterator::_init()
> >     @          0x16ecbce  doris::segment_v2::SegmentIterator::next_batch()
> >     @           0xfde23c  doris::BetaRowsetReader::next_block()
> >     @           0xf5b9b7  doris::CollectIterator::Level0Iterator::_refresh_current_row()
> >     @           0xf5bd12  doris::CollectIterator::add_child()
> >     @           0xf3a11e  doris::Reader::_capture_rs_readers()
> >     @           0xf3e6db  doris::Reader::init()
> >     @           0xf29800  doris::Merger::merge_rowsets()
> >     @           0xf18720  doris::Compaction::do_compaction_impl()
> >     @           0xf1ae4c  doris::Compaction::do_compaction()
> >     @           0xf0f7ec  doris::BaseCompaction::execute_compact_impl()
> >     @           0xf17004  doris::Compaction::execute_compact()
> >     @           0xe9f9ab  doris::Tablet::execute_compaction()
> >     @           0xe79c3f  _ZNSt17_Function_handlerIFvvEZN5doris13StorageEngine35_compaction_tasks_producer_callbackEvEUlvE0_E9_M_invokeERKSt9_Any_data
> >     @          0x120ac92  doris::ThreadPool::dispatch_thread()
> >     @          0x1202af8  doris::Thread::supervise_thread()
> >     @     0x7fddd1716ea5  start_thread
> >     @     0x7fddd1a298dd  __clone
> > ```
> 
> @qidaye Seems this bug is not same with morningman's? What version/commit id you are using?
> Maybe it has been resolved #5108

sorry, my mistake. I have pasted a wrong stack trace.
I've update the correct stack trace.

Thanks for pointing out.
